virtual_rainforest/core/utils.py

```diff
@@ -9,41 +9,59 @@
 from typing import Any
 
 import pint
-from numpy import datetime64, timedelta64
+from numpy import timedelta64
 
 from virtual_rainforest.core.exceptions import ConfigurationError, InitialisationError
 from virtual_rainforest.core.logger import LOGGER
 
 
-def extract_model_time_details(
-    config: dict[str, Any], model_name: str
-) -> tuple[datetime64, timedelta64]:
+def extract_update_interval(
+    config: dict[str, Any], lower_bound: str, upper_bound: str
+) -> timedelta64:
```

Comment:
Isn't this now largely a method of `BaseModel` that gets called in `BaseModel.__init__`? We should parse the `pint` Quantity for `update_interval` once from the config in setup in `main` and then it gets passed in to the BaseModel. So then we'd have this as:

```python
def _check_update_speed(self, update_interval: Quantity) -> Quantity:
```
and then in `BaseModel.__init__` do:

```python
self.update_interval = self._check_update_speed(update_interval)
```